File: fixtures/scale/ruby/large/lib/check_augustus.rb
require_relative 'augustus_encrypt'
require_relative 'augustus_decrypt'
require_relative 'corpus_additive'

def check_augustus
  corpus_additive.each do |text|
    sealed = augustus_encrypt(text, 8)
    opened = augustus_decrypt(sealed, 8)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_augustus.py
from cipherlab.ciphers.augustus import augustus_encrypt, augustus_decrypt
from cipherlab.specs.augustus_spec import augustus_spec
from cipherlab.corpus.corpus_additive import corpus_additive


def check_augustus():
    spec = augustus_spec()
    for text in corpus_additive():
        sealed = augustus_encrypt(text, spec["key"])
        opened = augustus_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "additive"
